docs/tx_lifecycle.md

```diff
@@ -0,0 +1,119 @@
+
+**See**:
+
+- [ArbOS](arbos/ArbOS)
+- [Geth](arbos/ArbOS)
+- [L1 pricing](arbos/L1_Pricing) / [L2 Gas](Gas)
+
+Note that RBlock assertions include claims about the state of the Outbox; if our transaction triggered any L2 to L1 messages, a RBlock will include an update to the Outbox to reflect its inclusion.
+
+**See**:
+
+- "The Outbox"
+
+#### 4a. RBlock is valid / goes unchallenged
+
+In the happy / common case, the validator asserted a valid RBlock, and over the course of the dispute window — 1 week on Arbitrum One — no other validators challenge it.
+
+#### 4b. Assertion is challenged!
+
+If two validators assert different RBlocks, only (at most) one of them can be valid, so they are put into a dispute.
+
+A dispute consists of two staked validators dissecting their disagreement down to a single L2 block, and then dissecting the sequence of VM instructions within this block down to a single OPCODE, then finally, executing this single operation. The underlying VM the Arbitrum uses is Wasm, or, more precicelsy, "WAVM." This is all refereed by contracts on L1.
+
+**See:**
+
+- [Challenges](proving/ChallengeManager)
+- [Wasm/WAVM](proving/WASMToWAVM)
+
+L1 contracts also keep track of the tree of all assertions; i.e., how many stakers are in disagreement, who is currently disputing with whom, etc. We refer to this level of Arbitrum's design architecture as its "rollup protocol."
+
+**See:**
+
+- "Rollup Protocol"
+
+#### ~ ~ ~ FINALITY CHECK: STILL THE SAME Ethereum-Equivalent Finality! ~ ~ ~
+
+Remember in phase 3 when said that once the L1 has committed to inputs, we can guarantee the L2 output? We meant it! Even during a dispute, Arbitrum nodes continue to execute and active validators continue to make assertions on the valid leaf in the state-tree; nothing that can happen in phase 4 has any effect on the L1-level finality we've already locked in at phase 3.
+
+### 5. RBlock is confirmed on L1
+
+Once any all disputes have been resolved and sufficient time has passed, our RBlock can be confirmed on L1 (any Ethereum account on L1 can confirm it). Upon confirmation, the Outbox root on L1 gets updated.
+
+#### ~ ~ ~ FINALITY CHECK: L2 to L1 Messages Executable on L1 ~ ~ ~
+
+If our client's transaction didn't include any L2 to L1 messages (i.e., withdrawals), phase 5 has no material affect on their transaction. If it did include an L2 to L1 transaction, it is only after confirmation that the message can be executed in the Outbox. Note that even before phase 5, the client has L1 finality on the _result_ of their L2 to L1 message, they just can't execute it yet; i.e., they're have a guarantee that they'll eventually be able to, e.g., finalize their withdrawal, they just can't actually claim their funds on L1 until the RBlock is confirmed.
```

Comment:
`i.e., withdrawals or cross-chain messages`?

Reply:
"as a common example" is intended, so should prob be "e.g." :)